# Get image attribute

# 1. Shape

## Get Grayscale image

In [1]:
import cv2
path = "I:\\openCV\\test.jpg"
# be careful of path and \\
image = cv2.imread(path,flags=0)
# check if the image load successfully or not 
if image is None:
    print("Fail to load")
else:
    cv2.imshow('image', image)
    cv2.waitKey(0)
    cv2.destroyAllWindows()

## Get the shape of a grayscale image

### We use image.shape (no parenthesis) to get how many rows and columns of pixels in a image

In [2]:
image.shape 

(320, 640)

Because this is a grayscale image, in that case the return value only have two parameters.

## Get BGR image

In [3]:
import cv2
path = "I:\\openCV\\test.jpg"
# be careful of path and \\
image1 = cv2.imread(path)
# check if the image load successfully or not 
if image1 is None:
    print("Fail to load")
else:
    cv2.imshow('image1', image1)
    cv2.waitKey(0)
    cv2.destroyAllWindows()

## Get the shape of a BGR image

In [4]:
image1.shape

(320, 640, 3)

Because this is a BGR image, in that case the return value have three parameters. The 3rd parameter is the number of color channel.

## 2. Number of pixels

### Grayscale size

In [5]:
image.size 

204800

### BGR size

In [6]:
image1.size

614400

We use image.size to get how many pixels in an image, the return value is the number of pixel points

The result comes from image.shape return value, for example, the grayscale's image.shape is (320, 640), so the image.size = 320*640 = 20480

The BGR's image1.shape is (320, 640, 3), so the image1.size = 320*640*3 = 61440

## 3. Get image type

In [8]:
image.dtype

dtype('uint8')

In [9]:
image1.dtype

dtype('uint8')